grid2op/multi_agent/subgridAction.py

```diff
@@ -26,7 +26,36 @@ def __init__(
                              actionClass=actionClass,
                              _extra_name=agent_name)
     
+    def _get_possible_action_types(self):
+        """Overrides an ActionSpace's method
+
+        Returns
+        -------
+        list
+            All possible action types
+        """
+        rnd_types = []
+        cls = type(self)
+        if self.n_line > 0: #TODO interco v0.1
+            if "set_line_status" in self.actionClass.authorized_keys:
+                rnd_types.append(cls.SET_STATUS_ID)
+            if "change_line_status" in self.actionClass.authorized_keys:
+                rnd_types.append(cls.CHANGE_STATUS_ID)
+            if "set_bus" in self.actionClass.authorized_keys:
+                rnd_types.append(cls.SET_BUS_ID)
+            if "change_bus" in self.actionClass.authorized_keys:
+                rnd_types.append(cls.CHANGE_BUS_ID)
```

Comment:
LOL !

Reply:
But yes of course if you put that, then you can sample action with the test that you did. It will never sample set_bu or change_bus action in the case there is no line.

Are you sure ?

So if you don't have line on your grid, you cannot set generator bus, nor load nor storage nor interco. Yep for sure, your agents will work awesome !


... THINK BEFORE YOU CODE !!!!!

Reply:
Corrected it